extensions/amp-analytics/0.1/test/vendor-requests.json

```diff
@@ -80,6 +80,11 @@
     "base": "https://scomcluster.cxense.com/Repo/rep.gif",
     "pageview": "https://scomcluster.cxense.com/Repo/rep.gif?ver=1&typ=pgv&sid=!siteId&ckp=_client_id_&loc=_source_url_&rnd=_random_&ref=_document_referrer_&ltm=_timestamp_&wsz=_screen_width_x_screen_height_&bln=_browser_language_&chs=_document_charset_&col=_screen_color_depth_&tzo=_timezone_&cp_cx_channel=amp"
   },
+  "deepbi": {
+    "base": "https://!endpoint/v1/pixel/!datasetKey/p.gif?accessKey=!accessKey&gdpr=!gdpr&page_href=_page_href_&referrer_href=_document_referrer_&page_title=_title_&amp=true",
```

Comment:
According to your deepbi.js config, `endpoint` and `gdpr` are both declared variables, and thus will be substituted here:
 '!endpoint' -> 'api.deep.bi'
'!gdpr' -> 'true' 

Additionally, you have 'page_href=${sourceUrl}' in your config. So that will turn into '_source_url' instead of 'page_href'. 

Applying these changes should fix the failing tests.

Reply:
Hello @micajuine-ho,

Thank you very much! I'd like to ask for your advice here, because we might misunderstood some requirements.
So, our default endpoint is `api.deep.bi`, but we'd like to let client to change it (some clients use 1st party endpoint to escape adblocking). So, we thought we'd make the endpoint a variable.
The same applies to `gdpr` parameter - without `gdpr=true` we are not collecting any cookies and by default clients must explicitly ask for the permission and then set `gdpr=true`.
Could you please advise us how to do it correctly?

For `page_href` - this is the name of parameter in our endpoint that accepts URL, we want it to hold the "true", non AMP URL value. I guess we are doing it incorrectly. Could you please advise us on how to fix it correctly?

THank you very much in advance!

Best regards,

Pawel K.

Reply:
Hi @pkalinowski,

The config that you submit (deepbi.js) can be thought of as the default. Clients can override the config (variables included) through their [inline config](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/amp-analytics.md#inline-or-remote-configuration). 

Since [variables get encoded](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/analytics-vars.md#-variables-supported-in-amp-analytics), you should not make the endpoint domain a variable. Instead, you should use the [request origin](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/amp-analytics.md#request-origin) property of the config. AKA, set the `requestOrigin` property to `https://api.deep.bi` in deepbi.js, and rely on publishers to override it in their inline config if they chose to.

For gdpr, it sounds like you expect publishers to handle setting `gdpr=true`. If this is the case, this should be doable using variables (nothing needs to change, unless you want gdpr to be default false). Additionally, you might want to look into the [data-consent-notification-id attribute](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/amp-analytics.md#valid-attributes-for-amp-analytics) that can be added to the amp-analytics tag.

For page_href, here is the documentation for [${sourceUrl}](https://github.com/ampproject/amphtml/blob/master/spec/amp-var-substitutions.md#source-url). It sounds like what you want, but remember that variables get encoded (see **Example value**). 

Please let me know if you need any more clarification. 

Best,
Mica

Reply:
Hello @micajuine-ho ,
Thank you very much for your explanation. We decided to go with change of request origin.
We do need to get more experience with AMP and we are going to rock with all the nice options it provides!

Reply:
Awesome! These test requests still need to be fixed to be inline with the changes you made to deepbi.js (such as `!endpoint` -> `api.deep.bi`, and the other issues I mentioned in my first response).

Reply:
Hello @micajuine-ho,
Sorry to bother you, but if there is anything we are missing here, please let me know, we are committed here to get our humble analytics going with AMP.